src/main/groovy/edu/oregonstate/mist/invencycle/resources/BikeResource.groovy

```diff
@@ -93,6 +94,40 @@ class BikeResource extends Resource {
     }
 
     /**
+     * PUT bike with ID
+     */
+    @PUT
+    @Path('{id: \\d+}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putByID (@PathParam('id') Integer id, @Valid Bike userBike) {
+        Response returnResponse
+        Integer idCheck = bikeDAO.idCheck(id)
+        Bike currentBike
+
+        if (!idCheck) {
+            returnResponse = badRequest("Bike ID does not exist. Use a POST request to create a new bike.").build()
```

Comment:
In your (!idCheck) method, I would just return out of it immediately. It is more of a small style comment. When I look towards the end of your method you are already three levels deep in identation / logic. One level less usually helps keep the code easier to read.

As long as you don't have long methods with a few return statements sprinkled in hidden (or hard to read) areas it's not hard to read to use return statements early in the method.